src/video/windows/SDL_windowswindow.c

```diff
@@ -669,12 +670,21 @@ void WIN_SetWindowIcon(_THIS, SDL_Window *window, SDL_Surface *icon)
 
     SDL_small_free(icon_bmp, isstack);
 
+    if (hicon == NULL) {
+        SDL_SetError("SDL_SetWindowIcon(): Unable to create a resource from the icon");
+        retVal = (int)GetLastError();
```

Comment:
I am not sure that:

1. We should return at this point or let `SendMessage` be executed
2. We should actually poll windows to return a proper error message.

Reply:
@slouken can help with those concerns better

Reply:
We definitely shouldn't return this error, we should return -1, but you can show the error number in the error message, like we do here:
```c
return SDL_SetError("GetClientRect() failed, error %08X", (unsigned int)GetLastError());
```